### PR TITLE
Fix test_ind_worker_queue by setting max_num_worker based on system resource

### LINKED ISSUE #63657: DISABLED test_ind_worker_queue (__main__.TestIndividualWorkerQueue)
https://app.circleci.com/pipelines/github/pytorch/pytorch/367226/workflows/e734b996-f8a8-466e-83b4-24973aff8a75/jobs/15494643

It seems worker freezing.
Potential solution is to set range based on CPU counts. https://github.com/pytorch/pytorch/blob/b9fc656cf26d60127bd695e4e5a7d27622f2563d/test/test_dataloader.py#L2280

cc @SsnL @VitalyFedyunin @ejguan

### PR DESCRIPTION
**Pull request by ejguan**
Fixes #63657

Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#63779 Fix test_ind_worker_queue by setting max_num_worker based on system resource**

Prevent freezing test for CI.

Confirm this test is flacky due to the limitation of CPU per machine. Did reproduce the hanging workers with ASAN build of PyTorch periodically.

After this PR, the number of worker will be limited based on OS.

Differential Revision: [D30494185](https://our.internmc.facebook.com/intern/diff/D30494185)

cc @SsnL @VitalyFedyunin @ejguan

### CODE REVIEW

**Comment by ejguan** [comment]
@ejguan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30494185).

**Comment by ejguan** [comment]
@ejguan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30494185).

**Comment by ejguan** [comment]
@ejguan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30494185).

File: test/test_dataloader.py [hunk 1]
```diff
@@ -2288,8 +2288,23 @@ def _run_ind_worker_queue_test(self, batch_size, num_workers):
...
+            cpu_count = os.cpu_count()
+            if cpu_count is not None:
+                # Use half number of CPUs
+                max_num_workers = cpu_count // 2
+
+        if max_num_workers is None:
+            max_num_workers = 1
+
         for batch_size in (8, 16, 32, 64):
-            for num_workers in range(1, 6):
```
Comment (VitalyFedyunin) [inline test/test_dataloader.py:2292]: nit: min(6, max_num_workers)

**Review by VitalyFedyunin — APPROVED** [review]

**Comment by ejguan** [comment]
@ejguan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30494185).

**Pull request closed without merging**